src/Simplify_Call.cpp

```diff
@@ -40,7 +40,9 @@ Expr Simplify::visit(const Call *op, ExprInfo *bounds) {
             }
 
             if (ib >= 0) {
-                if (ib < std::min(t.bits(), 64) - 1) {
+                int max_shift = std::min(t.bits(), 64);
+                if (t.is_int()) max_shift -= 1;
```

Comment:
Brace the if or do ```max_shift -= t.is_int(); // One less if there is a sign bit``` I'd also rename ```max_shift``` because it is not actually allowed. E.g. ```effective_type_bits``` or ```shift_too_large_bits```.

Reply:
Done.